src/internal/operators/first.ts

```diff
@@ -72,10 +72,10 @@ export function first<T>(
  * @method first
  * @owner Observable
  */
-export function first<T>(
+export function first<T, S extends T>(
   predicate?: ((value: T, index: number, source: Observable<T>) => boolean) | null,
-  defaultValue?: T
-): MonoTypeOperatorFunction<T> {
+  defaultValue?: S
+): OperatorFunction<T, S> {
```

Comment:
this doesn't seem right... the default value doesn't have to extend `T`, it could be anything.

`first<T, D> ` and return `:OperatorFunction<T, T|D>` seems more accurate, then type overloads for the more common cases.